### PR TITLE
user-guide: check-links presubmit

### PR DESCRIPTION
**Pull request by jbpratt**
Signed-off-by: jbpratt <jbpratt78@gmail.com>

### CODE REVIEW

**Comment by jbpratt** [comment]
Closes https://github.com/kubevirt/user-guide/issues/415

Hi @dhiller :smiley_cat: ,

I wanted to give this a shot (mostly to learn prow) and this is still WIP. I have a few questions around how this should actually work. Currently, I am building and pushing https://raw.githubusercontent.com/kubevirt/project-infra/main/images/kubevirt-user-guide/Dockerfile to my personal repository. This is because the `check_links` target tries to build the image (resulting in docker-in-docker) which I got around by copying various bits from existing jobs. After that I hit a `tty` error because the target is running with [`-t`](https://github.com/kubevirt/user-guide/blob/main/Makefile#L156). It was ultimately just easiest for me to do it this way.

Would it be possible to adding the above image to whatever process regularly builds images? (I would love to learn and help :smile_cat: )

I think it would be ideal to use the target directly, so if that is preferred please let me know any tips you may have.

**Comment by jbpratt** [comment]
```
project-infra  user-guide-check-links [+?]
❯ CONFIG_PATH=$(realpath github/ci/prow-deploy/files/config.yaml) JOB_CONFIG_PATH=$(realpath github/ci/prow-deploy/files/jobs/kubevirt/user-guide
/user-guide-presubmits.yaml) ./pj-on-kind.sh  check-links-user-guide
job=check-links-user-guide
CONFIG_PATH=/home/jbpratt/git/kubevirt/project-infra/github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml
JOB_CONFIG_PATH=/home/jbpratt/git/kubevirt/project-infra/github/ci/prow-deploy/files/jobs/kubevirt/user-guide/user-guide-presubmits.yaml
OUT_DIR=/tmp/prow/disks/prowjob-out   (Only used when creating a new kind cluster.)
KIND_CONFIG=   (Only used when creating a new kind cluster.)
NODE_DIR=/tmp/prow/disks/kind-node   (Only used when creating a new kind cluster.)
latest: Pulling from k8s-prow/mkpj
Digest: sha256:44c3234b7ec5aa93ed073b47c95dedc9b1d71aae991805557decc7dce29e2fb9
Status: Image is up to date for gcr.io/k8s-prow/mkpj:latest
gcr.io/k8s-prow/mkpj:latest
time="2022-01-29T11:17:55Z" level=warning msg="empty -github-token-path, will use anonymous github client"
time="2022-01-29T11:17:55Z" level=info msg="Throttle(0, 0, *)" client=github
PR Number: 516
time="2022-01-29T11:17:56Z" level=info msg="GetPullRequest(kubevirt, user-guide, 516)" client=github
time="2022-01-29T11:17:57Z" level=debug msg="GetPullRequest(kubevirt, user-guide, 516) finished" client=github duration=315.445619ms
latest: Pulling from k8s-prow/mkpod
Digest: sha256:ee083358f132f07fd710cdb4009013abe9e20434083641f85cf49302180a77aa
Status: Image is up to date for gcr.io/k8s-prow/mkpod:latest
gcr.io/k8s-prow/mkpod:latest
time="2022-01-29T11:17:58Z" level=info msg="Generated build-id for job." build-id=1487384519257362432
time="2022-01-29T11:17:58Z" level=info msg="Pod-utils configured for local mode. Instead of uploading to GCS, files will be copied to an output dir on the node." out-dir=/tmp/prow/disks/prowjob-out/check-links-user-guide/1487384519257362432
Applying pod to the mkpod cluster. Configure kubectl for the mkpod cluster with:
>  export KUBECONFIG='/home/jbpratt/.kube/kind-config-mkpod'
NAME                                   READY   STATUS     RESTARTS   AGE
1c53e592-80f5-11ec-9b2d-0242ac110003   0/2     Init:0/3   0          0s
1c53e592-80f5-11ec-9b2d-0242ac110003   0/2     Init:0/3   0          1s
1c53e592-80f5-11ec-9b2d-0242ac110003   0/2     Init:1/3   0          5s
1c53e592-80f5-11ec-9b2d-0242ac110003   0/2     Init:2/3   0          6s
1c53e592-80f5-11ec-9b2d-0242ac110003   0/2     PodInitializing   0          7s
1c53e592-80f5-11ec-9b2d-0242ac110003   2/2     Running           0          8s
1c53e592-80f5-11ec-9b2d-0242ac110003   0/2     Completed         0          28s
...
❯ k logs 1c53e592-80f5-11ec-9b2d-0242ac110003 test -f
mkdocs build -f mkdocs.yml -d site

Building site...
WARNING  -  Config value: 'site_favicon'. Warning: Unrecognised configuration name: site_favicon
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /home/prow/go/src/github.com/kubevirt/user-guide/site
INFO     -  Documentation built in 0.50 seconds

Checks html files for broken external links (This can take a few mins to run) ...
Running ["ImageCheck", "ScriptCheck", "LinkCheck"] on ["./site"] on *.html...


Checking 272 external links...
Ran on 49 files!


HTML-Proofer finished successfully.

Checks html files for broken internal links (This can take a few mins to run) ...
Running ["ImageCheck", "ScriptCheck", "LinkCheck"] on ["./site"] on *.html...


Ran on 49 files!


HTML-Proofer finished successfully.
```

**Comment by dhiller** [comment]
Hey @jbpratt, thanks for picking this up! We appreciate everyone helping in this area. I'll take a look at this later today.

**Comment by dhiller** [comment]
> Closes [kubevirt/user-guide#415](https://github.com/kubevirt/user-guide/issues/415)
> 
> Hi @dhiller smiley_cat ,
> 
> I wanted to give this a shot (mostly to learn prow) and this is still WIP. I have a few questions around how this should actually work. Currently, I am building and pushing https://raw.githubusercontent.com/kubevirt/project-infra/main/images/kubevirt-user-guide/Dockerfile to my personal repository. This is because the `check_links` target tries to build the image (resulting in docker-in-docker) which I got around by copying various bits from existing jobs. After that I hit a `tty` error because the target is running with [`-t`](https://github.com/kubevirt/user-guide/blob/main/Makefile#L156). It was ultimately just easiest for me to do it this way.
> 
> Would it be possible to add the above image to whatever process regularly builds and publishes images to quay? (I would love to learn and help smile_cat )
> 
> I think it would be ideal to use the target directly, so if that is preferred please let me know any tips you may have.
> 
> Edit:
> 
> So I went digging and found that the image already has a job
> 
> https://github.com/kubevirt/project-infra/blob/591e2e62062606514421ed38f0a957e2862c2aa9/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml#L148-L178
> but I can't seem to find the image.

Actually this is a known problem of new repositories with quay, they are initially created as private (quay folks is working on making the default switchable). I've made the repo public, you should be able to see it now.

Other than than, I think we can just take the job in like it is now and work on the remaining stuff in another PR. I'd leave that up to you to decide.

**Review by dhiller — CHANGES_REQUESTED** [review]
Hey @jbpratt, thank you for your work! Looks great already!

I've checked the job with [`phaino`](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/phaino/README.md) and it works fine. I'd just comment the cluster to run this on in.

File: github/ci/prow-deploy/files/jobs/kubevirt/user-guide/user-guide-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+presubmits:
+  kubevirt/user-guide:
+  - name: check-links-user-guide
+    always_run: true
```
[the remark this answers is not in the document]
Comment (jbpratt) [reply github/ci/prow-deploy/files/jobs/kubevirt/user-guide/user-guide-presubmits.yaml]: Uncommented line

**Comment by jbpratt** [comment]
Thanks for the review @dhiller , updated as requested. I think this should be ready to go!

**Review by dhiller — APPROVED** [review]
/approve

Thanks!

**Pull request merged**